scripts/test-e2e-kubernetes.sh

```diff
@@ -0,0 +1,108 @@
+  #
+  # Hint #1: you can use `scripts/start-docker-registry.sh`, but it requires
+  # manually preparing the environment to allow insecure registries, and it can
+  # also not work if you k8s cluster doesn't have network connectivity to the
+  # registry.
+  #
+  # Hint #2: if using with minikube, set `USE_MINIKUBE_CACHE` to `true` and use
+  # any value for `CONTAINER_IMAGE_REPO` (for instance, `vector-test` will do).
+  #
+  CONTAINER_IMAGE_REPO="${CONTAINER_IMAGE_REPO:?"You have to specify CONTAINER_IMAGE_REPO to upload the test image to."}"
+
+  # Assign a default test run ID if none is provided by the user.
+  TEST_RUN_ID="${TEST_RUN_ID:-"$(date +%s)-$(random-string)"}"
+
+  if [[ "${QUICK_BUILD:-"false"}" == "true" ]]; then
+    # Build in debug mode.
+    cargo build
+
+    # Prepare test image parameters.
+    VERSION_TAG="test-$TEST_RUN_ID"
+
+    # Prepare the container image for the deployment command and docker build.
+    CONTAINER_IMAGE="$CONTAINER_IMAGE_REPO:$VERSION_TAG-debug"
+
+    # Build docker image.
+    docker build --tag "$CONTAINER_IMAGE" -f skaffold/docker/Dockerfile target/debug
+  else
+    # Package a .deb file to build a docker container, unless skipped.
+    if [[ -z "${SKIP_PACKAGE_DEB:-}" ]]; then
+      make package-deb-x86_64 USE_CONTAINER="${PACKAGE_DEB_USE_CONTAINER:-"docker"}"
+    fi
+
+    # Prepare test image parameters.
+    VERSION_TAG="test-$TEST_RUN_ID"
+    BASE_TAG="debian"
+
+    # Build docker image with Vector - the same way it's done for releses. Don't
+    # do the push - we'll handle it later.
+    REPO="$CONTAINER_IMAGE_REPO" \
+      CHANNEL="test" \
+      BASE="$BASE_TAG" \
+      TAG="$VERSION_TAG" \
+      PUSH="" \
+      scripts/build-docker.sh
+
+    # Prepare the container image for the deployment command.
+    CONTAINER_IMAGE="$CONTAINER_IMAGE_REPO:$VERSION_TAG-$BASE_TAG"
+  fi
+fi
+
+if [[ -z "${SKIP_CONTAINER_IMAGE_PUBLISHING:-}" ]]; then
+  # Whether to use minikube cache to pass image to the k8s cluster.
+  # After we build vector docker image, instead of pushing to the remote repo,
+  # we'll be using `minikube cache` to make image available to the cluster.
+  # This effectively eliminates the requirement to have a docker registry, but
+  # it requires that we run against minikube cluster.
+  USE_MINIKUBE_CACHE="${USE_MINIKUBE_CACHE:-"false"}"
+
+  # Make the container image accessible to the k8s cluster.
+  if [[ "$USE_MINIKUBE_CACHE" == "true" ]]; then
+    minikube cache add "$CONTAINER_IMAGE"
+  else
+    docker push "$CONTAINER_IMAGE"
+  fi
+fi
+
+# Export the container image to be accessible from the deployment command.
+export CONTAINER_IMAGE
+
+# Set the deployment command for integration tests.
+export KUBE_TEST_DEPLOY_COMMAND="scripts/deploy-kubernetes-test.sh"
+
+# Prepare args.
+CARGO_TEST_ARGS=()
+if [[ -n "${SCOPE:-}" && "$SCOPE" != '""' ]]; then
+  CARGO_TEST_ARGS+=("$SCOPE")
+fi
+
+# Run the tests.
+cargo test \
```

Comment:
Really really happy to see I can run this on my localhost.

Reply:
It runs from localhost, AND you can use regular `kubectl` to debug it if something goes wrong, because the test itself uses that same `kubectl` to drive it's operations! No more headaches with "test fails, but how do I repro it and start tinkering with it".